src/address/mod.rs

```diff
@@ -329,6 +339,58 @@ where
     }
 }
 
+/// A weakly referenced counterpart to `Recipient<M>`
```

Comment:
As a user, I'd find some more docstring helpful here to explain what's the typical usecase of this weak recipient.
In particular, I found your mention of "risk of leaking objects" enlightening; it could mirrored and expanded in the docstring here.

Reply:
this doc-string is a reflection of the one for `WeakAddr`. You are right about the usecase example, IDK if it fits in here, but maybe an example or integration test would be good.

Reply:
The docs enhancement isn't absolutely necessary, but it'd be great if we could add a minimal example for it :)

Reply:
@hoodie thanks for the pair of examples, much appreciated!